src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,965 @@
+
+        /// <summary>
+        /// Parses a archived_book resource name in string form into a <see cref="ArchivedBookName"/>.
+        /// </summary>
+        /// <param name="archivedBookName">The archived_book resource name in string form. Must not be <c>null</c>.</param>
+        /// <returns>The parsed <see cref="ArchivedBookName"/> if successful.</returns>
+        public static ArchivedBookName Parse(string archivedBookName)
+        {
+            GaxPreconditions.CheckNotNull(archivedBookName, nameof(archivedBookName));
+            TemplatedResourceName resourceName = s_template.ParseName(archivedBookName);
+            return new ArchivedBookName(resourceName[0], resourceName[1]);
+        }
+
+        /// <summary>
+        /// Tries to parse archived_book resource name in string form into a <see cref="ArchivedBookName"/>/
+        /// </summary>
+        /// <remarks>
+        /// This method still throws <see cref="ArgumentNullException"/> if <paramref name="archivedBookName"/> is null,
+        /// as this would usually indicate a programming error rather than a data error.
+        /// </remarks>
+        /// <param name="archivedBookName">The archived_book resource name in string form. Must not be <c>null</c>.</param>
+        /// <param name="result">When this method returns, the parsed <see cref="ArchivedBookName"/>,
+        /// or <c>null</c> if parsing fails.</param>
+        /// <returns><c>true</c> if the name was parsed succssfully; <c>false</c> otherwise.</returns>
+        public static bool TryParse(string archivedBookName, out ArchivedBookName result)
+        {
+            GaxPreconditions.CheckNotNull(archivedBookName, nameof(archivedBookName));
+            TemplatedResourceName resourceName;
+            if (s_template.TryParseName(archivedBookName, out resourceName))
+            {
+                result = new ArchivedBookName(resourceName[0], resourceName[1]);
+                return true;
+            }
+            else
+            {
+                result = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Constructs a archived_book name from its component parts.
+        /// </summary>
+        /// <param name="archivePathId">The archivePath ID. Must not be <c>null</c>.</param>
+        /// <param name="bookIdId">The bookId ID. Must not be <c>null</c>.</param>
+        public ArchivedBookName(string archivePathId, string bookIdId)
+        {
+            ArchivePathId = GaxPreconditions.CheckNotNull(archivePathId, nameof(archivePathId));
+            BookIdId = GaxPreconditions.CheckNotNull(bookIdId, nameof(bookIdId));
+        }
+
+        /// <summary>
+        /// The archivePath ID. Never <c>null</c>.
+        /// </summary>
+        public string ArchivePathId { get; }
+
+        /// <summary>
+        /// The bookId ID. Never <c>null</c>.
```

Comment:
We could potentially be clever and omit the " ID" part if the name ends with "Id"